code/modules/modular_computers/computers/item/tablet.dm

```diff
@@ -42,9 +42,70 @@
 	has_variants = FALSE
 	device_theme = "syndicate"
 
+/obj/item/modular_computer/tablet/nukeops/Initialize()
+	. = ..()
+	set_light_color("#FF0000") //Syndicate likes it red
+
 /obj/item/modular_computer/tablet/nukeops/emag_act(mob/user)
 	if(!enabled)
 		to_chat(user, "<span class='warning'>You'd need to turn the [src] on first.</span>")
 		return FALSE
 	to_chat(user, "<span class='notice'>You swipe \the [src]. It's screen briefly shows a message reading \"MEMORY CODE INJECTION DETECTED AND SUCCESSFULLY QUARANTINED\".</span>")
 	return FALSE
+
+/// Borg Built-in tablet interface
+/obj/item/modular_computer/tablet/integrated
+	icon_state = "tablet-silicon"
+	has_light = FALSE //tablet light button actually enables/disables the borg lamp
+	comp_light_luminosity = 0
+	has_variants = FALSE
+	///Ref to the borg we're installed in. Set by the borg during our creation.
+	var/mob/living/silicon/robot/borgo
+	///IC log that borgs can view in their personal management app
+	var/list/borglog = list()
+
+//Makes the light settings reflect the borg's headlamp settings
+/obj/item/modular_computer/tablet/integrated/ui_data(mob/user)
+	. = ..()
+	.["has_light"] = TRUE
+	.["light_on"] = borgo?.lamp_enabled
+	.["comp_light_color"] = borgo?.lamp_color
+
+/obj/item/modular_computer/tablet/integrated/Initialize(mapload, container_borg)
+	. = ..()
+	borgo = container_borg
```

Comment:
```suggestion

/obj/item/modular_computer/tablet/integrated/Initialize(mapload, container_borg)
	. = ..()
	borgo = container_borg
//Makes the light settings reflect the borg's headlamp settings


/obj/item/modular_computer/tablet/integrated/Destroy()
	borgo = null
	return ..()


/obj/item/modular_computer/tablet/integrated/ui_data(mob/user)
	. = ..()
	.["has_light"] = TRUE
	.["light_on"] = borgo?.lamp_enabled
	.["comp_light_color"] = borgo?.lamp_color
```
* Clean references to avoid hard dels.
* Keep `Initialize()` and `Destroy()` at the top, right after the object definition, as they define the most basic behavior.